services-geocoding/src/main/java/com/mapbox/api/geocoding/v5/MapboxGeocoding.java

```diff
@@ -607,6 +628,22 @@ public MapboxGeocoding build() {
         && geocoding.limit() != null && !geocoding.limit().equals("1")) {
         throw new ServicesException("Limit must be combined with a single type parameter");
       }
+
+      if (intersectionStreets.size() == 2) {
+        if (!(geocoding.mode().equals(GeocodingCriteria.MODE_PLACES)
+                || geocoding.mode().equals(GeocodingCriteria.MODE_PLACES_PERMANENT))) {
+          throw new ServicesException("Geocoding mode must be " + GeocodingCriteria.MODE_PLACES
+                  + " or " + GeocodingCriteria.MODE_PLACES_PERMANENT + " for intersection search.");
+        }
+        if (TextUtils.isEmpty(geocoding.geocodingTypes())
+                || !geocoding.geocodingTypes().equals(GeocodingCriteria.TYPE_ADDRESS)) {
+          throw new ServicesException("Geocoding type must be set to "
+                  + GeocodingCriteria.TYPE_ADDRESS + " for intersection search.");
```

Comment:
Maybe the following instead?
```
"Geocoding type must be set to GeocodingCriteria.TYPE_ADDRESS for intersection search.");
```

What you have now will print out "Geocoding type must be set to address for intersection search." In reality, we want devs to actually pass through `GeocodingCriteria.TYPE_ADDRESS`.

Reply:
Thanks for the comments, I will fix it here. Also I am setting Geocoding type to GeocodingCriteria.TYPE_ADDRESS automatically when the `intersectionStreet()` is used.